assets/model-diagram.mmd

```diff
@@ -0,0 +1,11 @@
+erDiagram
+
+%% This is the source file for `model-diagram.svg`.
+%% Prerequisites: `npm i -g @mermaid-js/mermaid-cli svgo`
+%% Build command: `make model-diagram`
+
+"User agent" ||--|| "Storage shed" : "holds (local)"
+"Browsing session" ||--|| "Storage shed" : "holds (session)"
+"Storage shed" ||--o{ "Storage shelf" : "for each storage key"
+"Storage shelf" ||--|| "Storage bucket" : "holds (default)"
+"Storage bucket" ||--|{ "Storage bottle" : "for each storage endpoint"
```

Comment:
Shall we lowercase all these terms? I'm not entirely sure, but it looks a little off to me at the moment.